File: Data Structures & Algorithms/top-k-elements-in-list/submission-5.py
class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        counts = {}
        freq = [[] for _ in range(len(nums) + 1)]

        for num in nums:
            if num not in counts:
                counts[num] = 0
            counts[num] += 1
        for key, val in counts.items():
            freq[val].append(key)
        ans = []
        for i in range(len(nums), 0, -1):
            ans += freq[i]
            if len(ans) == k:
                return ans

File: Data Structures & Algorithms/longest-consecutive-sequence/submission-5.go
func longestConsecutive(nums []int) int {
    mp := make(map[int]int)
    ans := 0
    for _, num := range nums {
        if mp[num] > 0 {
            continue
        }
        left := mp[num - 1]
        right := mp[num + 1]
        mp[num] = left + right + 1
        ans = max(ans, mp[num])
        mp[num - left] = mp[num]
        mp[num + right] = mp[num]
    }
    return ans
}
